pkg/controller/operators/olm/operator.go

```diff
@@ -1893,3 +1917,66 @@ func (a *Operator) ensureLabels(in *v1alpha1.ClusterServiceVersion, labelSets ..
 	out, err := a.client.OperatorsV1alpha1().ClusterServiceVersions(out.GetNamespace()).Update(out)
 	return out, err
 }
+
+func (o *Operator) syncNamespace(obj interface{}) error {
+	ns, ok := obj.(*corev1.Namespace)
+	if !ok {
+		o.logger.Debugf("wrong type: %#v", obj)
+		return fmt.Errorf("casting Namespace failed")
+	}
+	namespace := ns.GetName()
+
+	logger := o.logger.WithFields(logrus.Fields{
+		"namespace": namespace,
+		"id":        queueinformer.NewLoopID(),
+	})
+
+	// get the set of sources that should be used for resolution and best-effort get their connections working
+	logger.Info("resolving namespace")
+
+	// Remove existing OperatorGroup labels
+	for label := range ns.GetLabels() {
+		if strings.HasPrefix(label, operatorGroupLabelPrefix) {
+			delete(ns.Labels, label)
+		}
+	}
+
+	// Get the list of OperatorGroups
+	operatorGroups, err := o.lister.OperatorsV1().OperatorGroupLister().OperatorGroups(metav1.NamespaceAll).List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	// Check if this Namespace requires the OperatorGroup Label
+	for _, og := range operatorGroups {
+		if containsString(og.Status.Namespaces, namespace) {
+			o.logger.WithFields(logrus.Fields{
+				"ogNamespace": og.GetNamespace(),
+				"ogName":      og.GetName(),
+			})
+			if ns.Labels == nil {
+				ns.Labels = make(map[string]string, 1)
+			}
+			ns.Labels[getOperatorGroupLabel(og)] = ""
+		}
+	}
+
+	// Update the Namespace
+	_, err = o.opClient.KubernetesInterface().CoreV1().Namespaces().Update(ns)
+	return err
+}
+
+func containsString(list []string, item string) bool {
+	for i := range list {
+		if list[i] == item {
+			return true
+		}
+	}
+	return false
+}
+
+// getOperatorGroupLabel returns a label that is applied to Namespaces to signify that the
+// namespace is a part of the OperatorGroup using selectors.
+func getOperatorGroupLabel(o *v1.OperatorGroup) string {
```

Comment:
nit: would this be prettier as a method on an OperatorGroup wrapper?

Reply:
Updated so the OperatorGroup struct has a GetLabel() func.